### PR TITLE
Fix split admittance for transfomer ratio different from 1

### PR DESCRIPTION
**Pull request by vidaldid-rte**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**
No

**What kind of change does this PR introduce?**
Corrects the formula to split addmittance on a transformer with ratio that is not one.

**What is the current behavior?**
The admittance is divided b two and one part is set on each one.
If the transformer ratio (ratedU2/ratedU1) is not one, this doesn't lead to the same transformer because the lef

**What is the new behavior (if this is a feature change)?**
Corrects the formula so that each transformer side has the same admittance but the transofmer  is not modified.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [X] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

**Comment by SylvestreSakti** [comment]
I added a paragraph in the documentation to explicit the used formula to split admittance. Do you agree with it ?

File: src/main/java/com/powsybl/openloadflow/network/impl/Transformers.java [hunk 1]
```diff
@@ -154,12 +154,14 @@ public static TapCharacteristics getTapCharacteristics(ThreeWindingsTransformer
...
+        // do not change g or b, then the splitted g1/g2 b1/b2 are not changed.
         double r = tapCharacteristics.getR() / zb;
         double x = tapCharacteristics.getX() / zb;
-        double g1 = (twtSplitShuntAdmittance ? tapCharacteristics.getG() / 2 : tapCharacteristics.getG()) * zb;
+        double g1 = (twtSplitShuntAdmittance ? tapCharacteristics.getG() * ratedRatio / (ratedRatio + 1) : tapCharacteristics.getG()) * zb;
         double g2 = twtSplitShuntAdmittance ? g1 : 0;
-        double b1 = (twtSplitShuntAdmittance ? tapCharacteristics.getB() / 2 : tapCharacteristics.getB()) * zb;
+        double b1 = (twtSplitShuntAdmittance ? tapCharacteristics.getB() * ratedRatio / (ratedRatio + 1) : tapCharacteristics.getB()) * zb;
         double b2 = twtSplitShuntAdmittance ? b1 : 0;
         double r1 = tapCharacteristics.getRatio() / baseRatio;
```
[the remark this answers is not in the document]
Comment (vidaldid-rte) [reply src/main/java/com/powsybl/openloadflow/network/impl/Transformers.java:166]: This is not the same thing
ratedratio is RatedU1/RatedU2 and is used to convert impedance move to the right (and with the assution that they are before the variable tap ideal transformer in the original pi model of a transformer)

baseRatio is to converted the ratio of the tap to the nominal referential of the transformer that depends on the voltage level. Baiscally it converts a leg' ratedU in per unit voltage for that voltage level.

File: src/test/java/com/powsybl/openloadflow/network/VoltageControlNetworkFactory.java [hunk 2]
```diff
@@ -413,6 +413,57 @@ public static Network createNetworkWithT2wt() {
...
+        Network network = VoltageControlNetworkFactory.createTransformerBaseNetwork("two-windings-transformer-control");
+
+        TwoWindingsTransformer t2wt = network.getSubstation("SUBSTATION").newTwoWindingsTransformer()
+                .setId("T2wT")
+                .setRatedU1(132.0)
+                .setRatedU2(33.0)
+                .setR(17.0)
+                .setX(10.0)
+                .setG(0.00573921028466483)
+                .setB(0.000573921028466483)
```
[the remark this answers is not in the document]
Comment (vidaldid-rte) [reply src/test/java/com/powsybl/openloadflow/network/VoltageControlNetworkFactory.java]: Done

File: src/test/java/com/powsybl/openloadflow/ac/TwtSplitShuntAdmittanceTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        assertEquals(lfTwt.getPiModel().getB1(), lfTwt.getPiModel().getB2());
+
+        // If we move B2 to the left, the result should be sames as TWT (in Per Unit)
+        double ratio = twt.getRatedU2() / twt.getRatedU1();
+        double bLeft = lfTwt.getPiModel().getB1() + lfTwt.getPiModel().getB2() / ratio;
+        assertEquals(twt.getB() * PerUnit.zb(twt.getTerminal2().getVoltageLevel().getNominalV()), bLeft, 1e-5);
+
+        assertNotEquals(0, lfTwt.getPiModel().getG2());
+        assertEquals(lfTwt.getPiModel().getG1(), lfTwt.getPiModel().getG2());
+        // If e move G2 to the left the result should be sames as TWT (in Per Unit)
```
[the remark this answers is not in the document]
Comment (vidaldid-rte) [reply src/test/java/com/powsybl/openloadflow/ac/TwtSplitShuntAdmittanceTest.java]: Done

File: src/test/java/com/powsybl/openloadflow/ac/TwtSplitShuntAdmittanceTest.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        List<LfNetwork> lfNetworks = new LfNetworkLoaderImpl().load(n, new LfTopoConfig(), params, ReportNode.NO_OP);
+        assertEquals(1, lfNetworks.size());
+
+        LfNetwork lfNetwork = lfNetworks.get(0);
+        LfBranch lfTwt = lfNetwork.getBranchById("T2wT");
+
+        assertNotEquals(0, lfTwt.getPiModel().getB2());
+        assertEquals(lfTwt.getPiModel().getB1(), lfTwt.getPiModel().getB2());
+
+        // If we move B2 to the left, the result should be sames as TWT (in Per Unit)
```
[the remark this answers is not in the document]
Comment (vidaldid-rte) [reply src/test/java/com/powsybl/openloadflow/ac/TwtSplitShuntAdmittanceTest.java]: Done

File: src/test/java/com/powsybl/openloadflow/ac/TwtSplitShuntAdmittanceTest.java [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    @Test
+    void testPiModelOnTransformer() {
+        // Test that the shunt split is correct
+        Network n = VoltageControlNetworkFactory.createNetworkWithVoltageReguulatingT2wtTapChangeRationOnly();
+
+        TwoWindingsTransformer twt = n.getTwoWindingsTransformer("T2wT");
+
+        // Create pi model array and splti transformer shunt admittance
```
[the remark this answers is not in the document]
Comment (vidaldid-rte) [reply src/test/java/com/powsybl/openloadflow/ac/TwtSplitShuntAdmittanceTest.java]: Done

File: docs/loadflow/loadflow.md [hunk 6]
```diff
@@ -17,6 +17,9 @@ equal to zero and $1$. In case of a branch with voltage or phase control, the $\
 ![Pi model](pi-model.svg){class="only-light"}
 ![Pi model](pi-model-dark-mode.svg){class="only-dark"}
 
+Note that transformers are modeled with a single admittance on one side (conductance $G$ and susceptance $B$). This admittance can be splitted into two parts (on both sides) with the [PowSyBl LoadFlow `twtSplitShuntAdmittance` Parameter](inv:powsyblcore:*:*#simulation/loadflow/configuration).
```
Comment (jeandemanged) [inline docs/loadflow/loadflow.md:20]: grammar: `can be splitted` -> `can be split`

**Review by jeandemanged — CHANGES_REQUESTED** [review]

File: docs/loadflow/loadflow.md (hunk 6 above)
Comment (SylvestreSakti) [reply docs/loadflow/loadflow.md:20]: Thanks, I forgot my irregular verbs :eyes:

**Pull request closed without merging**